src/transformers/models/vit/feature_extraction_vit.py

```diff
@@ -0,0 +1,255 @@
+                  :obj:`max_resolution` or to the maximum acceptable input resolution for the model if that argument is
+                  not provided.
+                * :obj:`False` or :obj:`'do_not_pad'` (default): No padding (i.e., can output a batch with images of
+                  different resolutions).
+            max_resolution (:obj:`int`, `optional`):
+                Controls the maximum resolution to use by one of the truncation/padding parameters. If left unset or
+                set to :obj:`None`, this will use the predefined model maximum resolution if a maximum resolution is
+                required by one of the truncation/padding parameters. If the model has no specific maximum input
+                resolution, truncation/padding to a maximum resolution will be deactivated.
+            pad_to_multiple_of (:obj:`int`, `optional`):
+                If set will pad the sequence to a multiple of the provided value. This is especially useful to enable
+                the use of Tensor Cores on NVIDIA hardware with compute capability >= 7.5 (Volta).
+            return_attention_mask (:obj:`bool`, `optional`):
+                Whether to return the pixel mask. If left to the default, will return the pixel mask according
+                to the specific image processor's default.
+                `What are pixel masks? <../glossary.html#attention-mask>`__
+            return_tensors (:obj:`str` or :class:`~transformers.tokenization_utils_base.TensorType`, `optional`):
+                If set, will return tensors instead of list of python floats. Acceptable values are:
+                * :obj:`'tf'`: Return TensorFlow :obj:`tf.constant` objects.
+                * :obj:`'pt'`: Return PyTorch :obj:`torch.Tensor` objects.
+                * :obj:`'np'`: Return Numpy :obj:`np.ndarray` objects.
+            verbose (:obj:`bool`, `optional`, defaults to :obj:`True`):
+                Whether or not to print more information and warnings.
+        """
+        # Input type checking for clearer error
+        assert (
+            isinstance(images, PIL.Image.Image)
+            or isinstance(images, np.ndarray)
+            or isinstance(images, torch.Tensor)
+            or (
+                (
+                    isinstance(images, (list, tuple))
+                    and (
+                        len(images) == 0
+                        or (
+                            isinstance(images[0], PIL.Image.Image)
+                            or isinstance(images[0], np.ndarray)
+                            or isinstance(images[0], torch.Tensor)
+                        )
+                    )
+                )
+            )
+        ), (
+            "Images must of type `PIL.Image.Image`, `np.ndarray` or `torch.Tensor` (single example),"
+            "`List[PIL.Image.Image]`, `List[np.ndarray]` or `List[torch.Tensor]` (batch of examples)."
```

Comment:
This is pretty unreadable. I think we should split it in several parts with a real if/elif/else test and raise a ValueError at the end.